services/controlplane.go

```diff
@@ -9,6 +9,15 @@ import (
 func RunControlPlane(controlHosts []hosts.Host, etcdHosts []hosts.Host, controlServices v1.RKEConfigServices) error {
 	logrus.Infof("[%s] Building up Controller Plane..", ControlRole)
 	for _, host := range controlHosts {
+
+		for _, role := range host.Role {
```

Comment:
Looks like we iterate over host roles a lot in the code to figure out if the host is worker vs control. Would it be helpful to have 2 booleans in the host struct - isControl,isWorker - and refer to them instead?